### PR TITLE
HPCC-17021 Improve result monitor query error handling

### PR DESCRIPTION
**Pull request by afishbeck**
Capture and fail on exceptions returned by query being monitored.  We
don't want any ambiguity about the state of what is stored in cassandra.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately

## Testing:

I generated new result monitors for my result monitoring test case.  I published, and ran the new
generated code.

### CODE REVIEW

**Comment by afishbeck** [comment]
@rpastrana please review.

File: esp/xslt/esdl2monitor.xslt [hunk 1]
```diff
@@ -197,6 +197,17 @@ END;
 </xsl:if>
   requestIn := DATASET([], the_requestLayout) : STORED ('<xsl:value-of select="$requestType"/>', FEW, FORMAT(FIELDWIDTH(100),FIELDHEIGHT(30), sequence(100)));
 
+  exceptionRec := RECORD
+    string10 Source {xpath('Source')};
+    integer2 Code {xpath('Code')};
+    string100 Message {xpath('Message')};
+  END;
+
+  soapout_rec := record
```
Comment (rpastrana) [inline esp/xslt/esdl2monitor.xslt]: very minor, but the record naming convention should be consistent.
exceptionRec vs soapout_rec

**Review by rpastrana — COMMENTED** [review]
@afishbeck one minor comment.

File: esp/xslt/esdl2monitor.xslt (hunk 1 above)
Comment (afishbeck) [inline esp/xslt/esdl2monitor.xslt]: Shows that I'm just deciding on my ECL conventions.

**Comment by afishbeck** [comment]
@rpastrana updated

**Comment by rpastrana** [comment]
@afishbeck looks reasonable

**Pull request merged**